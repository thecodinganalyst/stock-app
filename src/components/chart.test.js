import {fireEvent, render} from "@testing-library/react";
import Chart from "./chart";

describe("chart", () => {

    it('should render with provided dataSets and xTimestampSets props', () => {
        const dataSets = new Map([
            ['AAPL', [1, 2, 3]],
            ['GOOG', [4, 5, 6]]
        ]);
        const xTimestampSets = [1000, 2000, 3000];
        const { container } = render(<Chart dataSets={dataSets} xTimestampSets={xTimestampSets} />);
        expect(container).toMatchSnapshot();
    });

    it('should render with custom props', () => {
        const dataSets = new Map([
            ['AAPL', [1, 2, 3]],
            ['GOOG', [4, 5, 6]]
        ]);
        const xTimestampSets = [1000, 2000, 3000];
        const { container } = render(<Chart dataSets={dataSets} xTimestampSets={xTimestampSets} width={500} height={300} marginTop={10} marginRight={20} marginBottom={30} marginLeft={40} />);
        expect(container).toMatchSnapshot();
    });

})
