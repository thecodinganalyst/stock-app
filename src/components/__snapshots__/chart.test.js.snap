// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chart should render with custom props 1`] = `
<div>
  <div
    style="width: 500px; max-width: 100%; display: flex; flex-direction: column; align-items: center; justify-content: center;"
  >
    <svg
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 500 300"
      width="100%"
    >
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="260"
          pointer-events="all"
          width="440"
          x="40"
          y="10"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          shape-rendering="crispEdges"
          stroke="#fff"
          style="display: none;"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          style="display: none;"
        />
      </g>
      <g
        transform="translate(0, 270)"
      >
        <g
          fill="none"
          font-family="sans-serif"
          font-size="10"
          text-anchor="middle"
        >
          <path
            class="domain"
            d="M40.5,6V0.5H480.5V6"
            stroke="currentColor"
          />
          <g
            class="tick"
            opacity="1"
            transform="translate(84.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              07:50
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(150.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              07:55
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(216.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08 AM
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(282.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:05
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(348.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:10
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(414.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:15
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(480.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:20
            </text>
          </g>
        </g>
      </g>
      <g
        transform="translate(40, 0)"
      >
        <g
          fill="none"
          font-family="sans-serif"
          font-size="10"
          text-anchor="end"
        >
          <path
            class="domain"
            d="M-6,270.5H0.5V10.5H-6"
            stroke="currentColor"
          />
          <g
            class="tick"
            opacity="1"
            transform="translate(0,270.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              1.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,244.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              1.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,218.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              2.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,192.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              2.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,166.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              3.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,140.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              3.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,114.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              4.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,88.50000000000001)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              4.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,62.499999999999986)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              5.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,36.49999999999999)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              5.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,10.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              6.0
            </text>
          </g>
        </g>
      </g>
      <g>
        <path
          d="M40,270L260,218L480,166"
          fill="none"
          stroke="#1976d2"
          stroke-width="1.5"
        />
        <g
          fill="white"
          stroke="#1976d2"
          stroke-width="1.5"
        >
          <circle
            cx="40"
            cy="270"
            r="2.5"
          />
          <circle
            cx="260"
            cy="218"
            r="2.5"
          />
          <circle
            cx="480"
            cy="166"
            r="2.5"
          />
        </g>
      </g>
      <g>
        <path
          d="M40,114L260,62L480,10"
          fill="none"
          stroke="#9c27b0"
          stroke-width="1.5"
        />
        <g
          fill="white"
          stroke="#9c27b0"
          stroke-width="1.5"
        >
          <circle
            cx="40"
            cy="114"
            r="2.5"
          />
          <circle
            cx="260"
            cy="61.999999999999986"
            r="2.5"
          />
          <circle
            cx="480"
            cy="10"
            r="2.5"
          />
        </g>
      </g>
    </svg>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset Zoom
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`chart should render with provided dataSets and xTimestampSets props 1`] = `
<div>
  <div
    style="width: 800px; max-width: 100%; display: flex; flex-direction: column; align-items: center; justify-content: center;"
  >
    <svg
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 800 400"
      width="100%"
    >
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="340"
          pointer-events="all"
          width="720"
          x="40"
          y="20"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          shape-rendering="crispEdges"
          stroke="#fff"
          style="display: none;"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          style="display: none;"
        />
      </g>
      <g
        transform="translate(0, 360)"
      >
        <g
          fill="none"
          font-family="sans-serif"
          font-size="10"
          text-anchor="middle"
        >
          <path
            class="domain"
            d="M40.5,6V0.5H760.5V6"
            stroke="currentColor"
          />
          <g
            class="tick"
            opacity="1"
            transform="translate(112.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              07:50
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(220.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              07:55
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(328.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08 AM
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(436.50000000000006,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:05
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(544.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:10
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(652.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:15
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(760.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              08:20
            </text>
          </g>
        </g>
      </g>
      <g
        transform="translate(40, 0)"
      >
        <g
          fill="none"
          font-family="sans-serif"
          font-size="10"
          text-anchor="end"
        >
          <path
            class="domain"
            d="M-6,360.5H0.5V20.5H-6"
            stroke="currentColor"
          />
          <g
            class="tick"
            opacity="1"
            transform="translate(0,360.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              1.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,326.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              1.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,292.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              2.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,258.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              2.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,224.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              3.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,190.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              3.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,156.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              4.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,122.50000000000001)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              4.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,88.49999999999999)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              5.0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,54.49999999999999)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              5.5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,20.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              6.0
            </text>
          </g>
        </g>
      </g>
      <g>
        <path
          d="M40,360L400,292L760,224"
          fill="none"
          stroke="#1976d2"
          stroke-width="1.5"
        />
        <g
          fill="white"
          stroke="#1976d2"
          stroke-width="1.5"
        >
          <circle
            cx="40"
            cy="360"
            r="2.5"
          />
          <circle
            cx="400"
            cy="292"
            r="2.5"
          />
          <circle
            cx="760"
            cy="224"
            r="2.5"
          />
        </g>
      </g>
      <g>
        <path
          d="M40,156L400,88L760,20"
          fill="none"
          stroke="#9c27b0"
          stroke-width="1.5"
        />
        <g
          fill="white"
          stroke="#9c27b0"
          stroke-width="1.5"
        >
          <circle
            cx="40"
            cy="156"
            r="2.5"
          />
          <circle
            cx="400"
            cy="87.99999999999999"
            r="2.5"
          />
          <circle
            cx="760"
            cy="20"
            r="2.5"
          />
        </g>
      </g>
    </svg>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset Zoom
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
